src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -17,7 +17,7 @@
      * @param type - any class of Machine sub class. For example: Track.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Machine> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
```suggestion
    List<Machine> getAll(Class<? extends T> type);
```
it's better to use T in this case, that will make your code more flexible